### ISSUE TITLE
Maven installation fails - Dependencies issues 

### ISSUE THREAD

**Issue by petekun** [cloudant-labs/clouseau#18]
Maven installation fails with dependency issues. Below is the error.

"stdout_lines": [
        "[INFO] Scanning for projects...",
        "[INFO]                                                                         ",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Building clouseau 2.10.0-SNAPSHOT",
        "[INFO] ------------------------------------------------------------------------",
        "Downloading: http://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom",
        "",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] BUILD FAILURE",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Total time: 1.626 s",
        "[INFO] Finished at: 2018-03-19T15:02:08+00:00",
        "[INFO] Final Memory: 24M/962M",
        "[INFO] ------------------------------------------------------------------------",
        "[ERROR] Failed to execute goal on project clouseau: Could not resolve dependencies for project com.cloudant:clouseau:jar:2.10.0-SNAPSHOT: Failed to collect dependencies at com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Failed to read artifact descriptor for com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Could not transfer artifact com.boundary:scalang-scala_2.9.1:pom:0.28-cloudant2 from/to CloudantPublicRepo (http://maven.cloudant.com/repo/): Connect to maven.cloudant.com:80 [maven.cloudant.com/50.97.193.79] failed: Connection refused -> [Help 1]",
        "[ERROR] ",
        "[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.",
        "[ERROR] Re-run Maven using the -X switch to enable full debug logging.",
        "[ERROR] ",
        "[ERROR] For more information about the errors and possible solutions, please read the following articles:",
        "[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException"
    ]
}

**Comment by petekun** [comment]
@rnewson is it possible we return it to http://maven.cloudant.com? It's pointing to https://maven.cloudant.com causing the build to fail.

**Comment by rnewson** [comment]
um, I did a while ago, update?

**Comment by petekun** [comment]
Still failing

"stdout_lines": [
        "[INFO] Scanning for projects...",
        "[INFO]                                                                         ",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Building clouseau 2.10.0-SNAPSHOT",
        "[INFO] ------------------------------------------------------------------------",
        "Downloading: http://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom",
        "",
        "Downloading: https://repo.maven.apache.org/maven2/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom",
        "",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] BUILD FAILURE",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Total time: 2.240 s",
        "[INFO] Finished at: 2018-03-20T04:47:55+00:00",
        "[INFO] Final Memory: 25M/962M",
        "[INFO] ------------------------------------------------------------------------",
        "[ERROR] Failed to execute goal on project clouseau: Could not resolve dependencies for project com.cloudant:clouseau:jar:2.10.0-SNAPSHOT: Failed to collect dependencies at com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Failed to read artifact descriptor for com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Could not transfer artifact com.boundary:scalang-scala_2.9.1:pom:0.28-cloudant2 from/to CloudantPublicRepo (http://maven.cloudant.com/repo/): Connect to maven.cloudant.com:80 [maven.cloudant.com/50.22.242.80] failed: Connection refused -> [Help 1]",
        "[ERROR] ",
        "[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.",
        "[ERROR] Re-run Maven using the -X switch to enable full debug logging.",
        "[ERROR] ",
        "[ERROR] For more information about the errors and possible solutions, please read the following articles:",
        "[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException"
    ]
}

Still cant reach http://maven.cloudant.com/repo/

![image](https://user-images.githubusercontent.com/36202133/37636427-373a0308-2c13-11e8-8b9b-8c25934c022b.png)

**Comment by rnewson** [comment]
Hi, I got a bit confused here. I've put the url back to https://maven.cloudant.org/repo.

**Comment by petekun** [comment]
Hmm strange, still not working

  "stdout_lines": [
        "[INFO] Scanning for projects...",
        "[INFO]                                                                         ",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Building clouseau 2.10.0-SNAPSHOT",
        "[INFO] ------------------------------------------------------------------------",
        "Downloading: https://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom",
        "",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] BUILD FAILURE",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Total time: 3.378 s",
        "[INFO] Finished at: 2018-03-20T11:29:20+00:00",
        "[INFO] Final Memory: 25M/962M",
        "[INFO] ------------------------------------------------------------------------",
        "[ERROR] Failed to execute goal on project clouseau: Could not resolve dependencies for project com.cloudant:clouseau:jar:2.10.0-SNAPSHOT: Failed to collect dependencies at com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Failed to read artifact descriptor for com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Could not transfer artifact com.boundary:scalang-scala_2.9.1:pom:0.28-cloudant2 from/to CloudantPublicRepo (https://maven.cloudant.com/repo/): Connection reset -> [Help 1]",
        "[ERROR] ",
        "[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.",
        "[ERROR] Re-run Maven using the -X switch to enable full debug logging.",
        "[ERROR] ",
        "[ERROR] For more information about the errors and possible solutions, please read the following articles:",
        "[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException"
    ]
}

**Comment by rnewson** [comment]
can you visit `https://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom` directly? It works for me...

**Comment by petekun** [comment]
I can not sure why its failing. Below is the full error

 "changed": true,
    "cmd": "cd /home/couchdb/couchdb/clouseau && mvn -Dmaven.test.skip=true install",
    "delta": "0:00:02.650992",
    "end": "2018-03-20 15:03:58.811835",
    "failed": true,
    "invocation": {
        "module_args": {
            "_raw_params": "cd /home/couchdb/couchdb/clouseau && mvn -Dmaven.test.skip=true install",
            "_uses_shell": true,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": true
        }
    },
    "msg": "non-zero return code",
    "rc": 1,
    "start": "2018-03-20 15:03:56.160843",
    "stderr": "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset\nMar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: Retrying request to {s}->https://maven.cloudant.com:443\nMar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset\nMar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: Retrying request to {s}->https://maven.cloudant.com:443\nMar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset\nMar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute\nINFO: Retrying request to {s}->https://maven.cloudant.com:443",
    "stderr_lines": [
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset",
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: Retrying request to {s}->https://maven.cloudant.com:443",
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset",
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: Retrying request to {s}->https://maven.cloudant.com:443",
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://maven.cloudant.com:443: Connection reset",
        "Mar 20, 2018 3:03:58 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute",
        "INFO: Retrying request to {s}->https://maven.cloudant.com:443"
    ],
    "stdout": "[INFO] Scanning for projects...\n[INFO]                                                                         \n[INFO] ------------------------------------------------------------------------\n[INFO] Building clouseau 2.10.0-SNAPSHOT\n[INFO] ------------------------------------------------------------------------\nDownloading: https://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom\n\r[INFO] ------------------------------------------------------------------------\n[INFO] BUILD FAILURE\n[INFO] ------------------------------------------------------------------------\n[INFO] Total time: 1.564 s\n[INFO] Finished at: 2018-03-20T15:03:58+00:00\n[INFO] Final Memory: 25M/962M\n[INFO] ------------------------------------------------------------------------\n[ERROR] Failed to execute goal on project clouseau: Could not resolve dependencies for project com.cloudant:clouseau:jar:2.10.0-SNAPSHOT: Failed to collect dependencies at com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Failed to read artifact descriptor for com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Could not transfer artifact com.boundary:scalang-scala_2.9.1:pom:0.28-cloudant2 from/to CloudantPublicRepo (https://maven.cloudant.com/repo/): Connection reset -> [Help 1]\n[ERROR] \n[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.\n[ERROR] Re-run Maven using the -X switch to enable full debug logging.\n[ERROR] \n[ERROR] For more information about the errors and possible solutions, please read the following articles:\n[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException",
    "stdout_lines": [
        "[INFO] Scanning for projects...",
        "[INFO]                                                                         ",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Building clouseau 2.10.0-SNAPSHOT",
        "[INFO] ------------------------------------------------------------------------",
        "Downloading: https://maven.cloudant.com/repo/com/boundary/scalang-scala_2.9.1/0.28-cloudant2/scalang-scala_2.9.1-0.28-cloudant2.pom",
        "",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] BUILD FAILURE",
        "[INFO] ------------------------------------------------------------------------",
        "[INFO] Total time: 1.564 s",
        "[INFO] Finished at: 2018-03-20T15:03:58+00:00",
        "[INFO] Final Memory: 25M/962M",
        "[INFO] ------------------------------------------------------------------------",
        "[ERROR] Failed to execute goal on project clouseau: Could not resolve dependencies for project com.cloudant:clouseau:jar:2.10.0-SNAPSHOT: Failed to collect dependencies at com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Failed to read artifact descriptor for com.boundary:scalang-scala_2.9.1:jar:0.28-cloudant2: Could not transfer artifact com.boundary:scalang-scala_2.9.1:pom:0.28-cloudant2 from/to CloudantPublicRepo (https://maven.cloudant.com/repo/): Connection reset -> [Help 1]",
        "[ERROR] ",
        "[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.",
        "[ERROR] Re-run Maven using the -X switch to enable full debug logging.",
        "[ERROR] ",
        "[ERROR] For more information about the errors and possible solutions, please read the following articles:",
        "[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException"
    ]
}

**Comment by rnewson** [comment]
Are you sure you aren't having other networking/firewall issues preventing your outbound requests? Try the url in my last comment with curl or wget on the same machine.

**Comment by petekun** [comment]
I doubt, I've installed other packages, this is the only one bringing issues. I'm puzzled.

**Comment by rnewson** [comment]
have you tried the curl command yet?

**Comment by petekun** [comment]
Yeah I can curl

 sudo curl -k https://maven.cloudant.com/repo/
<html>
<head><title>Index of /repo/</title></head>
<body bgcolor="white">
<h1>Index of /repo/</h1><hr><pre><a href="../">../</a>
<a href="com/">com/</a>                                               30-Jan-2017 21:08                   -
<a href="jsr166y/">jsr166y/</a>                                           30-Jan-2017 21:08                   -
<a href="org/">org/</a>                                               30-Jan-2017 21:08                   -
</pre><hr></body>
</html>

**Comment by rnewson** [comment]
hm, ok (no idea why you sudo'ed that, that's a bad habit you should break).

have you configured maven to use a proxy maybe? (~/.m2/settings.xml iirc).

**Comment by petekun** [comment]
Cool will not use sudo :-) I haven't configured any proxies. Find below my settings.xml

`
<?xml version="1.0" encoding="UTF-8"?>




<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
  

  

  

  
  <pluginGroups>
    
  </pluginGroups>

  
  <proxies>
    
  </proxies>

  
  <servers>
    

    
  </servers>

  
  <mirrors>
    
  </mirrors>

  
  <profiles>
    

    
  </profiles>

  
</settings>
`

**Comment by petekun** [comment]
Can we try switching it http://maven.cloudant.com/repo/ instead of https?

**Comment by rnewson** [comment]
nope, because we've closed port 80 across our whole infrastructure and I'm not going to reverse that here. my maven install can contact https://maven.cloudant.com/repo and pull down all the artifacts. I don't understand why yours can't. :(

**Comment by petekun** [comment]
Hmm did a fresh install of the OS and I'm still encountering the same issue. Strange....it uses the same repo for Ubuntu 16.04 Xenial?

**Comment by rnewson** [comment]
yup, same repo, it's the same jar for every OS. I'm not really sure what to suggest you try next.

**Comment by pgj** [comment]
This issue is no longer relevant so now I am closing it.
